### PR TITLE
Add Flink logging receiver

### PR DESCRIPTION
**Pull request by JonathanWamsley**
Example output

```json
{
  "insertId": "148gda2f1jvvvu",
  "jsonPayload": {
    "source": "org.apache.flink.runtime.entrypoint.ClusterEntrypoint",
    "message": "        [] -  Starting StandaloneSessionClusterEntrypoint (Version: 1.14.4, Scala: 2.11, Rev:895c609, Date:2022-02-25T11:57:14+01:00)"
  },
  "resource": {
    "type": "gce_instance",
    "labels": {
      "instance_id": "8710493455276707297",
      "project_id": "otel-agent-dev",
      "zone": "us-central1-a"
    }
  },
  "timestamp": "2022-05-26T17:12:02.616Z",
  "severity": "INFO",
  "labels": {
    "compute.googleapis.com/resource_name": "flink-debian2"
  },
  "logName": "projects/otel-agent-dev/logs/flink",
  "receiveTimestamp": "2022-05-26T17:12:22.426334126Z"
}
```

### CODE REVIEW

File: docs/flink.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Apache Flink
+
+## Logs
+
+Flink logs contain the following fields in the [`LogEntry`](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) for Client, Jobmanager and Taskmanagers:
+
+| Field | Type | Description |
+| ---   | ---- | ----------- |
+| `jsonPayload.source` | string | Module and/or thread  where the log originated |
+| `jsonPayload.logger` | string | Name of the logger where the log originated |
```
Comment (ridwanmsharif) [inline docs/flink.md:10]: I don't see this field in the expected logs or in the parse regex. Are we missing something?

**Review by ridwanmsharif — APPROVED** [review]
LGTM modulo a small note

File: docs/flink.md (hunk 1 above)
Comment (JonathanWamsley) [reply docs/flink.md]: mb. removed `jsonPayload.logger`

File: integration_test/third_party_apps_data/applications/flink/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+public_url: "https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party/flink"
```
Comment (sophieyfang) [inline integration_test/third_party_apps_data/applications/flink/metadata.yaml:1]: This shouldn't be set as this is a 404 non existent link.

When google technical writer has the documentation, we will set this public_url in post agent release jobs :)

Comment (ridwanmsharif) [reply integration_test/third_party_apps_data/applications/flink/metadata.yaml:1]: Oh great catch! Will fix this as part of the flink metrics PR. Also made a pass through the other PRs that added this and commented on them too.

**Pull request merged**